src/Tester.ts

```diff
@@ -878,4 +883,788 @@ export class Tester {
+                    return null;
+            }
+            catch(e){
+                console.log("error: " + e);
+            }
+        }
+        /**
+         * Simple function to create HTTP(s) request options from given parameters.
+         */
+        function createRequestOptions(url: URL, op: string): object{
+            return {
+                hostname: url.hostname,
+                path: url.pathname,
+                port: url.port,
+                headers:{},
+                method: op
+            }
+        }
+        /**
+         * Returns the related form of the InteractionAffordance with given op.
+         */
+        function getForm(op: string, forms: Array<any>){
+
+            for (var i = 0; i < forms.length; i++){
+                if (forms[i]['op'].includes(op))
+                    return forms[i];
+            }
+            return null;
+        }
+        switch(scheme){
+            case "basic":
+                var location: string; // The 'in' parameter of the TD Spec.
+                report.scheme = 'basic';
+
+                if(!td['securityDefinitions'][schemeName]['in']){ // Default value.
+                    location = 'header';
+                }
+                else{
+                    location = td['securityDefinitions'][schemeName]['in'];
+                }
+
+                if (td['properties'] != undefined){ // Properties exist.                    
+                    const properties: any = Object.values(td['properties']);
+
+                    for (var i=0; i < properties.length; i++){
+                        let property: any = properties[i];
+
+                        // Creating propertyReport with the name of the property.
+                        report.createPropertyReport(Object.keys(td['properties'])[i]);
+                        report.propertyReports[i].createSecurityReport();
+
+                        if (!property.writeOnly){ // Property can be read, supposedly?
+                            try{
+                                // First tries to 'readproperty'.
+                                var form = getForm('readproperty', property.forms);
+
+                                // Check if the interaction has a different security scheme.
+                                if (form['security'] != undefined){
+                                    if (Array.isArray(form['security'])){
+                                        if (!form['security'].includes(schemeName))
+                                            throw "Testing multiple security schemas are not currently available.";
+                                    }
+                                    else if (form['security'] != schemeName)
+                                        throw "Testing multiple security schemas are not currently available.";
+                                }
+                                
+                                var propertyURL: URL = new URL(form['href']);
```

Comment:
there is an assumption of having all forms with http which needs to be tested first.

Reply:
Hocam, could you elaborate on that, please? It is stated in `README` that testing for vulnerabilities _currently_ only work for HTTP(s) requests.

Reply:
Yes of course, but there is no check being done to limit that. So what happens with a CoAP Thing, total chaos or a warning that it is not possible?